From ecbe652fd9c8d010f7f208860981682b8107f22f Mon Sep 17 00:00:00 2001
From: Kyle Fazzari <git@status.e4ward.com>
Date: Tue, 18 Sep 2012 18:20:32 -0400
Subject: [PATCH] UBUNTU: SAUCE: input: Cypress PS/2 Trackpad fix disabling
 tap-to-click

Fix "Disable tap-to-click doesn't affect mouse button zones" by making
sure the specialized taps coming from the mouse zones don't turn into
mouse clicks.

BugLink: http://launchpad.net/bugs/1048816

Signed-off-by: Kyle Fazzari <git@status.e4ward.com>
Signed-off-by: Kamal Mostafa <kamal@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/input/mouse/cypress_ps2.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/input/mouse/cypress_ps2.c b/drivers/input/mouse/cypress_ps2.c
index 45afe8f..3a572ec 100644
--- a/drivers/input/mouse/cypress_ps2.c
+++ b/drivers/input/mouse/cypress_ps2.c
@@ -693,6 +693,14 @@ static int cypress_parse_packet(const unsigned char packet[],
 		}
 	}
 
+	/* This is only true if one of the mouse buttons were tapped.
+	 * Make sure it doesn't turn into a click. The regular tap-to-
+	 * click functionality will handle that on its own. If we don't
+	 * do this, disabling tap-to-click won't affect the mouse button
+	 * zones. */
+	if (report_data->tap)
+		report_data->left = 0;
+
 	if (report_data->contact_cnt <= 0)
 		return 0;
 
-- 
1.8.0

